qutebrowser/completion/models/sortfilter.py

```diff
@@ -140,21 +133,33 @@ def filterAcceptsRow(self, row, parent):
         if parent == QModelIndex() or not self.pattern:
             return True
 
-        try:
-            return self.srcmodel.custom_filter(self.pattern, row, parent)
-        except NotImplementedError:
-            for col in self.srcmodel.columns_to_filter:
-                idx = self.srcmodel.index(row, col, parent)
-                if not idx.isValid():
-                    # No entries in parent model
-                    continue
-                data = self.srcmodel.data(idx)
-                if not data:
-                    continue
-                elif self.pattern_re.search(data):
-                    return True
+        # first check the cache
```

Comment:
No precise numbers since I have no idea how to measure the performance of `filterAcceptsRow`, which is called internally by Qt... It doesn't help that much when a lot of entries pass through the filter, but helps a lot when all entries are filtered out - this happens quite a lot in `UrlCompletionModel`, e.g. writing a new web search query. I think that even the current code lags considerably on each keypress when there are many entries in the history and you write a new search query...